src/OpenTK/Platform/X11/X11DisplayDevice.cs

```diff
@@ -173,6 +173,14 @@ private bool QueryXRandR(List<DisplayDevice> devices)
                 XRRScreenSize[] sizes = FindAvailableResolutions(screen);
                 foreach (XRRScreenSize size in sizes)
                 {
+                    // Calculate the scaling factor for this resolution
+                    // Use a default 1,1 in case the reported physical size is zero / undefined
+                    Vector2 scaleFactor = new Vector2(1, 1);
+                    if (size.MWidth != 0 && size.MHeight != 0)
+                    {
+                        scaleFactor = new Vector2((size.Width * 25.4f / size.MWidth) / 96.0f, (size.Height * 25.4f / size.MHeight) / 96.0f);
```

Comment:
I don't have time for a full review but it looks pretty good! Will do a full review in a few days.

One little comment is that generally I would prefer if the parentheses where for every priority e.g. `(a * b) / c` instead 
of `a * b / c`. Maybe even `(a * b) / (c * d)` in this case would make sense.
It would also be nice to have a comment describing where the `25.4f` and `96.0f` comes from.

Reply:
Up to you with the parenthesis.

IMHO they're superfluous (basic BIDMAS) & Resharper agrees, but meh :)

Reply:
In my opinion it's an extra layer of uncertainty that can just be avoided with just two more characters.
It doesn't reduce readability that much for people that are used to math notation, but it does increase readability for people that are not as math litterate.